src/main/webapp/app/shared/metis/forwarded-message/forwarded-message.component.ts

```diff
@@ -0,0 +1,114 @@
+import { AfterViewInit, ChangeDetectorRef, Component, ElementRef, OnChanges, OnInit, SimpleChanges, ViewChild, inject, input, output } from '@angular/core';
+import { faShare } from '@fortawesome/free-solid-svg-icons';
+import { Post } from 'app/entities/metis/post.model';
+import { AnswerPost } from 'app/entities/metis/answer-post.model';
+import { Posting } from 'app/entities/metis/posting.model';
+import dayjs from 'dayjs';
+import { Conversation } from 'app/entities/metis/conversation/conversation.model';
+
+@Component({
+    selector: 'jhi-forwarded-message',
+    templateUrl: './forwarded-message.component.html',
+    styleUrls: ['./forwarded-message.component.scss'],
+})
+export class ForwardedMessageComponent implements OnChanges, AfterViewInit, OnInit {
+    readonly faShare = faShare;
+    sourceName: string | undefined = '';
+    originalPostDetails = input<Posting>();
+    postingIsOfToday: boolean;
+    todayFlag?: string;
+    readonly onNavigateToPost = output<Posting>();
+    @ViewChild('messageContent') messageContent!: ElementRef;
+    isContentLong: boolean = false;
+    showFullForwardedMessage: boolean = false;
+    maxLines: number = 5;
+    private cdr = inject(ChangeDetectorRef);
+    private conversation: Conversation | undefined;
+    private isAnswerPost: boolean = false;
+    protected viewButtonVisible: boolean = false;
+
+    ngOnInit() {
+        this.isAnswerPost = 'post' in this.originalPostDetails()!;
+        this.conversation = this.isAnswerPost ? (this.originalPostDetails() as AnswerPost).post?.conversation : (this.originalPostDetails() as Post).conversation;
+        this.updateSourceName();
+        this.isChannel();
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
```

Comment:
Does ngOnChanges still work with input signals? I think we have to use effect({}) to react to changes here.